clippy_lints/src/use_self.rs

```diff
@@ -220,15 +218,35 @@ struct UseSelfVisitor<'a, 'tcx> {
 
 impl<'a, 'tcx> Visitor<'tcx> for UseSelfVisitor<'a, 'tcx> {
     fn visit_path(&mut self, path: &'tcx Path, _id: HirId) {
+        if path.segments.len() >= 2 {
+            let last_but_one = &path.segments[path.segments.len() - 2];
+            if last_but_one.ident.name != kw::SelfUpper {
+                let enum_did = match path.res {
+                    Res::Def(DefKind::Variant, variant_did) =>
```

Comment:
NIT: Please use `def_id` instead of `did`. (https://github.com/rust-lang/rust/issues/49509#issuecomment-497689795; I want to enforce this by a internal rustc lint, but didn't get to that yet)